### PR TITLE
Major code/config cleanup, improved documentation

### LINKED ISSUE #302: Setup readthedocs for Gumby
We should have a readthedocs page where we bundle all information about Gumby. This should contain some basic guides i.e. how to run an experiment using Tribler and Dispersy.

### LINKED ISSUE #309: Make PR test subjobs parallelizable
Right now the subjobs of the PR tester are in conflict with each other (they try to modify the same working tree).
To fix this I have the tests running sequentially instead of in parallel.

Ideally the subjobs would be completely independent and able to run in parallel though.

### LINKED ISSUE #444: Refactor the argument of get_peer to int
Currently, the `get_peer` method in the `IPv8OverlayExperimentModule` class takes a string to indicate the ID of a specific peer. This is most likely because the keys in the `all_vars` dict are of type `str`. It would be more convenient to refactor this argument to an integer.

### LINKED ISSUE #490: Fix the IPv8 tracker functionality in Gumby
In our current experiments, we manually connect peers with each other, resulting in a static network. Recently, the tracker functionality in Gumby broke down. It would be nice to fix this, so we can also run experiments with more realistic peer discovery/boostrapping.

### LINKED ISSUE #493: Refactor the IPv8 tracker into a separate module
Currently, the tracker functionality is [part of the experiment startup sequence](https://github.com/Tribler/gumby/blob/devel/gumby/runner.py#L161). For organisational purposes, we could move starting/managing the tracker to a separate module.

### LINKED ISSUE #495: The isolated community launcher is setting the `master_peer` variable instead of `community_id`
The isolated community launcher is setting the `master_peer` variable of the community instead of the `community_id` variable. This makes the current isolated community launcher useless.

### PR DESCRIPTION
**Pull request by devos50**
This PR contains a massive cleanup of the Gumby code. One can see the different commits to get an overview of the refactoring efforts in this PR. **Note that I included everything in a single PR so I could also overhaul the PR pipeline while refactoring. For organisational purposes, everything has been split into smaller commits as much as possible**. Some significant changes:
- Moved the IPv8 tracker functionality to a separate module.
- Removed (unused) functionality to run a 'remote' experiment (might be re-added later in a different form).
- Refactored `launch_scenario.py`.

The documentation has also been improved. In particular, I've written four tutorials that elaborate on the different aspects of Gumby experiments (e.g., DAS5 deployment and IPv8 integration). A readthedocs has been setup, which is accessible [here](https://gumby.readthedocs.io/en/latest/) after this PR has been merged. I've made a separate documentation job based on this PR which can be found [here](https://gumby-fork.readthedocs.io/en/latest/).

Schermafbeelding 2021-07-19 om 14 17 22

I also improved the PR pipeline. The PR pipeline now also includes a flake8 check, and runs the experiments that are described in the tutorial. Running these experiments will hopefully help to quickly detect whether someone has broken core functionality of Gumby. I've included some logic so that the experiment fails when the experiment does not deliver the expected artifacts.

Schermafbeelding 2021-07-19 om 14 19 45

Fixes #302, fixes #309, fixes #444, fixes #490, fixes #493, fixes #495

### CODE REVIEW

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

File: docs/tutorials/experiment_tutorial_1.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
+Running a Simple Experiment with Gumby
+======================================
+
+In this first tutorial, we will run a very simple experiment using the functionality that Gumby provides.
+This experiment is executed locally and spawns two independent CPU-intensive instances (processes) that execute
+the `yes` command for ten seconds.
```
Comment (qstokkink) [inline docs/tutorials/experiment_tutorial_1.rst]:
```suggestion
the ``yes`` command for ten seconds.
```

Comment (devos50) [reply docs/tutorials/experiment_tutorial_1.rst]: Thanks, fixed! 👍

**Review by qstokkink — APPROVED** [review]
That's all I had for you, looks good 👍

**Comment by devos50** [comment]
Thanks for the review! 👍

**Pull request merged**